client/my-sites/email/email-providers-stacked-comparison/email-provider-stacked-card/email-provider-generic-form.tsx

```diff
@@ -0,0 +1,868 @@
+		);
+
+	const [ alternativeEmailFieldTouched, setAlternativeEmailFieldTouched ] = useState( false );
+	const [ firstNameFieldTouched, setFirstNameFieldTouched ] = useState( false );
+	const [ fullNameFieldTouched, setFullNameFieldTouched ] = useState( false );
+	const [ lastNameFieldTouched, setLastNameFieldTouched ] = useState( false );
+	const [ mailBoxFieldTouched, setMailBoxFieldTouched ] = useState( false );
+	const [ passwordFieldTouched, setPasswordFieldTouched ] = useState( false );
+
+	const hasAlternativeEmailError = alternativeEmailFieldTouched && null !== alternativeEmailError;
+	const hasMailBoxError = mailBoxFieldTouched && null !== mailBoxError;
+	const hasFirstNameError = firstNameFieldTouched && null !== firstNameError;
+	const hasFullNameError = fullNameFieldTouched && null !== fullNameError;
+	const hasLastNameError = lastNameFieldTouched && null !== lastNameError;
+	const hasPasswordError = passwordFieldTouched && null !== passwordError;
+
+	const emailAddressLabel = translate( 'Email address' );
+
+	const renderSingleDomain = () => {
+		return (
+			<LabelWrapper label={ emailAddressLabel }>
+				<FormTextInputWithAffixes
+					value={ mailBox }
+					isError={ hasMailBoxError }
+					onChange={ ( event: ChangeEvent< HTMLInputElement > ) => {
+						onUserValueChange( 'mailBox', event.target.value.toLowerCase() );
+					} }
+					onBlur={ () => {
+						setMailBoxFieldTouched( wasValidated );
+					} }
+					onKeyUp={ onReturnKeyPress }
+					prefix={ isRtl ? `\u200e@${ selectedDomainName }\u202c` : null }
+					suffix={ isRtl ? null : `\u200e@${ selectedDomainName }\u202c` }
+				/>
+			</LabelWrapper>
+		);
+	};
+
+	const renderMultiDomain = () => {
+		return (
+			<LabelWrapper label={ emailAddressLabel }>
+				<div className="email-provider-generic-form__multi-container">
+					<FormTextInput
+						value={ mailBox }
+						isError={ hasMailBoxError }
+						onChange={ ( event: ChangeEvent< HTMLInputElement > ) => {
+							onUserValueChange( 'mailBox', event.target.value.toLowerCase() );
+						} }
+						onBlur={ () => {
+							setMailBoxFieldTouched( wasValidated );
+						} }
+						onKeyUp={ onReturnKeyPress }
+					/>
+
+					<DomainsSelect
+						domains={ domains }
+						onChange={ ( event ) => {
+							onUserValueChange( 'domain', event.target.value );
+						} }
+						value={ selectedDomainName }
+					/>
+				</div>
+			</LabelWrapper>
+		);
+	};
+
+	return (
+		<div className={ classNames( 'email-provider-generic-form__new-user' ) }>
+			<FormFieldset className="email-provider-generic-form__form-fieldset">
+				{ ! hiddenFields.some( ( field ) => field === GENERIC_EMAIL_FORM_FULL_NAME_FIELD ) && (
+					<div className="email-provider-generic-form__new-user-name-container">
+						<LabelWrapper label={ translate( 'Full name' ) }>
+							<FormTextInput
+								autoFocus={ autoFocus } // eslint-disable-line jsx-a11y/no-autofocus
+								value={ firstName }
+								maxLength={ 60 }
```

Comment:
It feels like it might be better to have these limits driven by some sort of config that is used here and in the validation checks.